pkg/controller/actions/status/releases/action_fetch_releases_status.go

```diff
@@ -0,0 +1,113 @@
+package releases
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/blang/semver/v4"
+	"github.com/opendatahub-io/opendatahub-operator/v2/apis/common"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/operator-framework/api/pkg/lib/version"
+	"gopkg.in/yaml.v3"
+	"path/filepath"
+)
+
+const (
+	ComponentMetadataFilename = "component_metadata.yaml"
+)
+
+// ComponentReleasesMeta represents the metadata for releases in the component_metadata.yaml file.
+// It contains a list of releases associated with a specific component.
+type ComponentReleasesMeta struct {
+	Releases []ComponentReleaseMeta `yaml:"releases,omitempty"`
+}
+
+// ComponentReleaseMeta represents the metadata of a single release within the component_metadata.yaml file.
+// It includes the release name, version, and repository URL.
+type ComponentReleaseMeta struct {
+	Name    string `yaml:"name,omitempty"`
+	Version string `yaml:"version,omitempty"`
+	RepoURL string `yaml:"repoUrl,omitempty"`
+}
+
+type Action struct {
+	labels map[string]string
```

Comment:
maybe add an option to set the location of the file, and fallback to the computed one if nor provided so you can easily unit test the action